### DISCUSSION TITLE
Discrepancy in Duration Accuracy Compared to Google Maps

### DISCUSSION THREAD

**Discussion by ak4zh** [adayush/olamaps-python#4] [General]
I have noticed a significant discrepancy in the duration results returned by Olamaps when compared to Google Maps. While the distance calculations are more or less the same, the duration estimates are considerably off.

Here is a specific example to illustrate the issue:

**Test 1:**
- **Origin:** 12.9716, 77.5946 (Bangalore, India)
- **Destination:** 13.0827, 80.2707 (Chennai, India)

**Olamaps Results:**
- Distance: 330 km
- Duration: 5 hours 43 minutes

**Google Maps Results:**
- Distance: 346 km
- Duration: 6 hours 40 minutes

**Test 2:**
- **Origin:** 28.6139, 77.2090 (Delhi, India)
- **Destination:** 11.0168, 76.9558 (Coimbatore, Tamil Nadu, India)

**Olamaps Results:**
- Distance: 2382 km
- Duration: 32 hours 31 minutes

**Google Maps Results:**
- Distance: 2424 km
- Duration: 42 hours

**Comment by adayush** [comment]
Really interesting. In the first test distance is 5% less but the duration is 15% less. Similarly in the second test, distance is 1.2% less but the duration is 23% less

**Comment by ak4zh** [comment]
It seems OLA currently does not have accuracy regarding traffic, road conditions etc.

Origin: 22.78277,86.139776
Destination: 20.740841,78.527476

- **OSRM Project:** 12 hours 45 minutes
- **OLA:** 14 hours 39 minutes
- **HERE Maps:** 15 hours
- **Google:** 21 hours

I have a transport application where I need more accurate duration so I am doing some analysis of duration on kmph basis for a temporary solution until OLA improves on that. I understand this is unreliable way to calculate the duration. As per my test on approx 1000 different co-ordinates

OLA average speed seems somewhere between 60-70 kmph
Google average speed around 45-55 kmph

**Reply by ak4zh** [reply]
Tested a free more maps api providers:

- [HERE Maps](https://platform.here.com/portal/) - results are identical to OLAMAPS
- [OSRM Project](https://map.project-osrm.org) - shows even less duration than OLAMAPS

All providers returns distance similar to google map distance.

**Reply by ak4zh** [reply]
For now I am using this formula to get approx duration which is approximately equal to google maps duration.

```python
distance_km = api_distance / 1000
duration_fifty_kmph = distance_km / 50 * 3600
duration_seconds = max(api_duration, duration_fifty_kmph)
```
